app/root.tsx

```diff
@@ -30,12 +32,24 @@ export const meta: MetaFunction = () => {
     return [{ title: 'ReClaim: Home Goods Store' }];
 };
 
-export async function loader() {
-    return {
-        ENV: {
-            WIX_CLIENT_ID: process?.env?.WIX_CLIENT_ID,
+export async function loader({ request }: LoaderFunctionArgs) {
+    const { wixEcomTokens, setCookie } = await initializeSession(request);
```

Comment:
Relationship between `initializeSession` and `initializeEcomApi` look a bit tangled to me.
`initializeEcomApi` expects tokens returned by `initializeSession`. At the same time `initializeSession` internally calls `initializeEcomApi` to generate visitor tokens.

Reply:
it works exactly as you described. will try to somehow make it less coupled